arborio/jsonio.cpp

```diff
@@ -0,0 +1,317 @@
+#include <fstream>
+#include <iomanip>
+#include <map>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include "arbor/util/any_visitor.hpp"
+#include "arborio/jsonio.hpp"
+#include "json_helpers.hpp"
+
+namespace arb {
+void to_json(nlohmann::json& j, const cable_cell_parameter_set& params) {
+    if (auto tempK = params.temperature_K) j["celsius"] = tempK.value() - 273.15;
+    if (auto Vm = params.init_membrane_potential) j["Vm"] = Vm.value();
+    if (auto Ra = params.axial_resistivity) j["Ra"] = Ra.value();
+    if (auto cm = params.membrane_capacitance) j["cm"] = cm.value();
+    for (auto ion: params.ion_data) {
```

Comment:
Since I cannot advertise this feature enough ;)

```
for  (const auto&[name, data]: params.ion_data) {
```

In any case, this should be a `const &` to avoid a copy.

Reply:
I will make the change.